x-pack/plugins/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -37,10 +39,25 @@ import type {
   GetFullAgentPolicyResponse,
   GetFullAgentConfigMapResponse,
   GetFullAgentManifestResponse,
+  BulkGetAgentPoliciesResponse,
 } from '../../../common/types';
-import { defaultIngestErrorHandler } from '../../errors';
+import { defaultIngestErrorHandler, AgentPolicyNotFoundError } from '../../errors';
 import { createAgentPolicyWithPackages } from '../../services/agent_policy_create';
 
+async function assignAgents(esClient: ElasticsearchClient, agentPolicies: AgentPolicy[]) {
```

Comment:
nit: the name is a bit confusing, we could call this function `populateAssignedAgentsCount`